hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -333,6 +341,11 @@ public LegacyReplicationManager(final ConfigurationSource conf,
         .setDBTransactionBuffer(scmhaManager.getDBTransactionBuffer())
         .setRatisServer(scmhaManager.getRatisServer())
         .setMoveTable(moveTable).build();
+
+    inflightMoveScannerExecutor = Executors.newSingleThreadExecutor(
```

Comment:
Thanks for the review. Let me think about it. My current idea is to shutdown the `ExecutorService` when stopping the `ReplicationManager` and reinitializing the `ExecutorService` in the `start`. Let me write more extensive tests to check this since `notifyStatusChanged` is run in Ratis and if any uncaught exception is thrown it might stop the whole `StateMachine`

Reply:
I think another issue is that `ReplicationManager#stop` will not replicate it to the other SCMs. Meaning that one SCM might have `running` flag `false`, while the other has `running` flag `true`. If for example we transfer leadership from the one with stopped RM to the one with running RM, the `notifyStatusChanged` will still trigger the `inflightMoveScannerExecutor`.

In my opinion, for now we can let `inflightMoveScannerExecutor` to send some replicate commands even if it's stopped.

Reply:
Yes, we can stop the thread only when the SCM stop.

Reply:
Will the `ReplicationManager#stop` be called only when the SCM shutdown or the user send "stop replicationManager" command, right?
So we may can let the inflightMoveScannerExecutor stop in the `ReplicationManager#stop`

Reply:
Yes, `ReplicationManager#stop` is called either in SCM shutdown through (`SCMServiceManager#stop`) or through the "stop replicationManager" command.

The reason I decided to now shutdown the `inflightMoveScannerExecutor` since once the `ExecutorService` is shutdown, we cannot submit any new task anymore. Therefore, we need to recreate this `ExecutorService` every time in `start`. However, managing this `ExecutorService`s lifecycles with the different Raft states might have some edge cases and error prone (which might cause `RejectedExecutionException` or ` NullPointerException`), this might terminate the `StateMachineUpdater`. Therefore, I try to make it simple and just let the `ExecutorService` to stop when SCM is shutdown (through the `ExecutorService` finalizer mechanism).

Reply:
OK, understand.